### PR TITLE
fix reduce

### PR DESCRIPTION
**Pull request by andicuko**
- when the query has a group by and doesn't have aggregation functions

related to this [issue](https://github.com/Qrlew/qrlew/issues/79#issuecomment-1670904822).

### CODE REVIEW

**Comment by andicuko** [comment]
Several files have been changed following `cargo fmt` command.

The actual changes are in `src/relation/builders.rs` and in `src/sql/relation.rs` to add a test.

File: src/multiplicity/mod.rs [hunk 1]
```diff
@@ -765,6 +767,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
```
Comment (ngrislain) [inline src/multiplicity/mod.rs:770]: remove all these

File: src/relation/builder.rs [hunk 2]
```diff
@@ -153,9 +153,7 @@ impl<RequireInput> MapBuilder<RequireInput> {
 
     /// Add a group by
     pub fn group_by(mut self, expr: Expr) -> Self {
-        self.split = self
-            .split
-            .map_last_reduce(|reduce| reduce.and(Split::group_by(expr)));
+        self.split = self.split.and(Split::group_by(expr.into()).into());
```
Comment (ngrislain) [inline src/relation/builder.rs:156]: Are you sure?

File: src/sql/relation.rs [hunk 3]
```diff
@@ -907,6 +907,28 @@ mod tests {
         println!("query = {q}");
     }
 
+    #[test]
+    fn test_reduce_with_only_group_by_columns() {
```
Comment (ngrislain) [inline src/sql/relation.rs:911]: Are we sure this is not working now without the change?

Is it working with piled map reduce: R, MR, RM, MRM, RMR

**Review by ngrislain — APPROVED** [review]

**Pull request merged**